src/main/java/persistence/entity/EntityManagerImpl.java

```diff
@@ -0,0 +1,42 @@
+package persistence.entity;
+
+import jdbc.JdbcTemplate;
+import jdbc.RowMapperImpl;
+import persistence.sql.dml.DmlQueryBuilder;
+
+import java.util.List;
+import java.util.Map;
+
+public class EntityManagerImpl implements EntityManager {
+    private final DmlQueryBuilder queryBuilder;
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public EntityManagerImpl(DmlQueryBuilder queryBuilder, JdbcTemplate jdbcTemplate) {
+        this.queryBuilder = queryBuilder;
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public <T> T findById(Class<T> clazz, Long id) {
+        String selectQuery = queryBuilder.buildSelectQuery(clazz, List.of(Map.of("id", id)));
```

Comment:
List<Map...> 이런 타입으로 사용하니까 가독성이 떨어진다는 단점이 있는 거 같아요.
차라리 getDefaultSelectQuery().where().build() 이런식으로 접근하는 게 나았을까? 싶은 생각도 들었는데 (다 구현하고 난 뒤에야..) 어떻게 생각하시나요?

Reply:
`getDefaultSelectQuery().whereEqual().build()`와 같은 방법을 사용해본다면 쿼리와 관련된 클래스 의존성이 `EntityManger`에 다시 생길 것 같아요! 🤔 

지금의 구조는 `buildSelectQuery`는 모든 select 쿼리를 만들 수 있다는 유연성과 확장성을 얻었습니다.
하지만 이런 유연성과 확장성을 가져가기 위해서는 `List<Map...>`과 같은 파라미터나 그에 더해  where절에 사용될 Class 정보 등 더 많은 메서드 파라미터를 요구하게 될 것 같아요!

여기서는 이제 트레이드 오프의 문제라고 생각해요!
1. 쿼리 관련 클래스에 대한 의존성과 복잡도를 가지고 여러 select 쿼리를 유연하게 만들 것인가
2. 쿼리 관련 클래스에 대한 의존성과 복잡도를 가지지 않고 select 쿼리를 제한적으로 만들 것인가

제 개인적인 의견으로는 JPA 요구사항에서는 아래의 예시와 같이 select 메서드를 만드는 것이 어떨까 생각이드네요!
```suggestion
        String selectQuery = queryBuilder.selectQueryById(clazz, id);
```

실제 JPA 구현에서도 EntityManager는 id를 통한 조회만 지원을 하고 있는데요!
JPA의 쿼리 메서드와 같은 기능들이 위와 같은 문제도 함께 해결하기 위해 노력하지 않았을까 생각이 드네요! :)

한번 고민해보시고 희원님의 기준대로 구현을 마무리해보시면 좋을 것 같습니다 😄

Reply:
반영했습니다! 말씀 듣고 spring data jpa랑 hibernate sessionImpl같은 이런저런 구현 코드들을 더 찾아보게 된 거 같아요 😄 
확실히 entity manager관점에선 id로 조회하는 기능으로 마무리 짓는 게 나을 거 같습니다. ㅎㅎ